src/Caching/StackExchangeRedis/src/RedisCache.cs

```diff
@@ -190,13 +190,20 @@ private void Connect()
             {
                 if (_cache == null)
                 {
-                    if (_options.ConfigurationOptions != null)
+                    if(_options.ConnectionMultiplexerFactory == null)
                     {
-                        _connection = ConnectionMultiplexer.Connect(_options.ConfigurationOptions);
+                        if (_options.ConfigurationOptions != null)
+                        {
+                            _connection = ConnectionMultiplexer.Connect(_options.ConfigurationOptions);
+                        }
+                        else
+                        {
+                            _connection = ConnectionMultiplexer.Connect(_options.Configuration);
+                        }
                     }
                     else
                     {
-                        _connection = ConnectionMultiplexer.Connect(_options.Configuration);
+                        _connection = _options.ConnectionMultiplexerFactory() as ConnectionMultiplexer;
```

Comment:
Could we avoid this `as` cast?

Reply:
@pranavkm we can if the `ConnectionMultiplexerFactory` func is declared as `Func<ConnectionMultiplexer>`? Would that be preferred?

Reply:
Instead, let's make `_connection` a `IConnectionMultiplexer` instead.